src/inmanta/server/services/compilerservice.py

```diff
@@ -226,6 +227,84 @@ async def _run_compile_stage(self, name: str, cmd: list[str], cwd: str, env: dic
                 # The process is still running, kill it
                 sub_process.kill()
 
+    async def ensure_compiler_venv(self) -> None:
```

Comment:
to be able to log, this needs to be async

1. Do I need to run the file operations on a thread?  (they block the ioloop)
2. Do I need to run the venv creation on a thread? (blocks the ioloop)